code/modules/shuttle/shuttle.dm

```diff
@@ -184,6 +187,28 @@
 	highlight("#f00")
 	#endif
 
+/obj/docking_port/stationary/proc/load_roundstart()
+	if(json_key)
+		var/sid = SSmapping.config.shuttles[json_key]
+		var/datum/map_template/shuttle/D = SSmapping.shuttle_templates[sid]
+		if(!istype(D))
+			CRASH("Bad json_key [json_key] for stationary port [src]")
+
+		roundstart_template = D
+
+	if(ispath(roundstart_template))
+		for(var/sid in SSmapping.shuttle_templates)
+			var/datum/map_template/shuttle/D = SSmapping.shuttle_templates[sid]
+			if(D.type == roundstart_template)
+				roundstart_template = D
+				break
+
+	if(istype(roundstart_template))
```

Comment:
`else if` or `return`

Reply:
I don't think you understand the flow of control here.

Reply:
ah I see now that it's not a local var, this is still a duplicate istype if you loaded from json so instead take out the istype check for crash above and move it down here to be `if(!istype(roundstart_template)) CRASH`

Reply:
But we want to be clear on whether it's a bad json_key or a bad typepath.. (how can we have a bad typepath though)